### DISCUSSION TITLE
encrypting nonce between client and server

### DISCUSSION THREAD

**Discussion by nmarandi** [jedisct1/libsodium#1197] [Q&A] [answered]
Is there any way to encrypt nonce between client and server?

I used [Key exchange](https://libsodium.gitbook.io/doc/key_exchange) and [Authenticated encryption](https://libsodium.gitbook.io/doc/public-key_cryptography/authenticated_encryption).

For decryption, I need to send nonce alongside the message. Is there any way to encrypt nonce?

**Comment by jedisct1** [comment, marked as answer]
Bellare recently published a paper about this: https://eprint.iacr.org/2019/624.pdf

But maybe you should explain the problem you are trying to solve first.

With 192-bit nonces, you can safely use random nonces instead of a counter. A middle box would learn nothing from such nonces.

**Reply by nmarandi** [reply]
Thank you for your fast reply. 
I will read the paper you sent. 
Mostly, I wanted to know whether encrypting nonce is needed or not. 
Currently, I use [randombytes_buf](https://libsodium.gitbook.io/doc/generating_random_data) to generate a nonce. 
So I can assume that sending nonce without encryption is ok.
Thanks!

**Comment by jedisct1** [comment]
Nonces don't have to be secret. They just have to be different for each message.
